emailsender/cmd/app/main.go

```diff
@@ -0,0 +1,74 @@
+// Package main for email sender service
+package main
+
+import (
+	"context"
+	"flag"
+	"net/http"
+	"os"
+	"os/signal"
+
+	"github.com/gorilla/mux"
+
+	"golang.org/x/sys/unix"
+
+	"github.com/golang/glog"
+	"github.com/stackrox/acs-fleet-manager/emailsender/config"
+)
+
+func main() {
+
+	// This is needed to make `glog` believe that the flags have already been parsed, otherwise
+	// every log messages is prefixed by an error message stating the flags haven't been
+	// parsed.
+	_ = flag.CommandLine.Parse([]string{})
+
+	// Always log to stderr by default, required for glog.
+	if err := flag.Set("logtostderr", "true"); err != nil {
+		glog.Info("unable to set logtostderr to true.")
+	}
+
+	cfg, err := config.GetConfig()
+	if err != nil {
+		glog.Fatal(err)
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), cfg.StartupTimeout)
+	defer cancel()
+
+	// base router
+	router := mux.NewRouter()
+
+	// example handler
+	router.HandleFunc("/test", func(rw http.ResponseWriter, req *http.Request) {
+		glog.Info("called /test endpoint")
+	})
+
+	server := http.Server{Addr: cfg.ServerAddress, Handler: router}
+
+	go func() {
+		glog.Info("Creating api server...")
+		if cfg.EnableHTTPS {
+			if err := server.ListenAndServeTLS(cfg.HTTPSCertFile, cfg.HTTPSKeyFile); err != http.ErrServerClosed {
+				glog.Fatalf("HTTPS API ListenAndServe error: %v", err)
+			}
+		} else {
+			if err := server.ListenAndServe(); err != http.ErrServerClosed {
+				glog.Fatalf("HTTP API ListenAndServe error: %v", err)
+			}
+		}
```

Comment:
I think this could be simpler.

Why are you filtering the error for http.ErrServerClosed?
If that's important you could also add it add the if for the log in my suggestion.

```suggestion
		var err error
		if cfg.EnableHTTPS {
			err := server.ListenAndServeTLS(cfg.HTTPSCertFile, cfg.HTTPSKeyFile)
		} else {
			err := server.ListenAndServe()
		}
		glog.Fatalf("ListenAndServer error: %v", err)
```

Reply:
I've applied your suggestion but returned `ErrServerClosed` filter. The reason is that closing server by interrupt signal is expected and normal shut down so such flow should not throw a error. Anything else which stops the server should be handled as an error